public/app/plugins/datasource/prometheus/datasource.ts

```diff
@@ -100,8 +100,11 @@ export class PrometheusDatasource extends DataSourceApi<PromQuery, PromOptions>
     this.directUrl = instanceSettings.jsonData.directUrl;
     this.resultTransformer = new ResultTransformer(templateSrv);
     this.ruleMappings = {};
-    this.languageProvider = new PrometheusLanguageProvider(this);
     this.customQueryParameters = new URLSearchParams(instanceSettings.jsonData.customQueryParameters);
+
+    if (!instanceSettings.jsonData.disableLanguageProvider) {
+      this.languageProvider = new PrometheusLanguageProvider(this);
```

Comment:
I think we should change the type of `languageProvider` to `languageProvider?` to reflect this. It necessary right now because of not running strict but will ease some fixes later when we do.

Reply:
Another thing is that this will disable both the metric dropdown but also all autocomplete. Autocomplete also suggest static things like functions which I think do not need server request. So question is if want to do it that way (could make sense some people already asked about switch for autocomplete in general) or we want to keep the autocomplete functionality where possible?

Reply:
About the type change - yes, thanks! I think I'll switch on strict mode while working just to catch stuff like this in the future. ;)

Reply:
Well, I for one would hope that this is just a temporary solution anyway but @davkal what do you think? Should I invest a bit more time here and not simply cut off the `languageProvider` completely?

Reply:
Yes good point! We don't want do disable the autocomplete entirely (e.g., for function names), but only the parts that use the API for metric and label lookups. (The name of the setting label "Disable metric lookup" is accurate.)